### PR TITLE
[WIP] feat: support retrying when calling metrics api

### PR DESCRIPTION
**Pull request by danbosnichill**
TESTING=unit tests

### CODE REVIEW

**Comment by danbosnichill** [comment]
@scottmcmaster - this was an old commit I had from April.  I'm clearing out some of my local commits now.  Let me know if you want this in here in this way.

I held off previously on merging this because I might want to push this into ApiClients that are used and provide default implementations of the ApiClients.

Let me know.

File: src/client.test.ts [hunk 1]
```diff
@@ -1486,6 +1489,8 @@ describe('deliver', () => {
       expect(deliveryClient.mock.calls.length).toBe(1);
       expect(metricsClient.mock.calls.length).toBe(1);
     });
+
+    // TODO -
```
Comment (scottmcmaster) [inline src/client.test.ts:1493]: What's this one for?

**Review by scottmcmaster — COMMENTED** [review]
The way the TS SDK is set up, I would lean in favor of pushing it to ApiClients. But I suppose the same could be said for timeouts? I could go either way. Maybe leave it out unless we have near-term plans to employ it?

**Comment by danbosnichill** [comment]
I'll close the PR.  I agree with moving this into the ApiClient.

**Pull request closed without merging**